ouroboros-consensus/src/Ouroboros/Consensus/Node/DbLock.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Ouroboros.Consensus.Node.DbLock (
+    DbLocked (..)
+  , dbLockFile
+  , lockDB
+  ) where
+
+import           Control.Concurrent
+import           Control.Monad (void)
+import           Data.Text (Text)
+import           Ouroboros.Consensus.Storage.FS.API
+import           Ouroboros.Consensus.Storage.FS.API.Types
+import           Ouroboros.Consensus.Util.ResourceRegistry
+
+import           Control.Exception
+import           System.FileLock
+
+-- | This can run from multiple processes concurrently, without any locking.
+-- It creates the db directory and its lock file, if any of them doesn't exist
+-- already.
+createDBFolder :: Monad m => HasFS m h -> m ()
+createDBFolder hasFS = do
+    createDirectoryIfMissing hasFS True root
+    void $ hOpen hasFS pFile (WriteMode AllowExisting)
+  where
+    root  = mkFsPath []
+    pFile = fsPathFromList ["dblock"]
+
+-- | We try to lock the db multiple times, before we give up and throw an
```

Comment:
As I said in https://github.com/input-output-hk/ouroboros-network/issues/1903#issuecomment-609640922, I wouldn't retry this until we know we need it.

Reply:
Let's wait until we're sure: https://github.com/input-output-hk/ouroboros-network/issues/1903#issuecomment-609660069

Reply:
If we use a timeout on a blocking lock acquire, we don't need retries.